### DISCUSSION TITLE
Is there a recommended MySQL client in Rust ecosystem?

### DISCUSSION THREAD

**Discussion by billy1624** [pingcap/tidb#35152] [Ideas]
Hello, I'm coming from Rust and found this amazing project! I want to use it with Rust and I'm looking for a TiDB client in Rust. Is there any recommended one?

**Comment by lance6716** [comment]
https://docs.pingcap.com/tidb/stable/mysql-compatibility

you should be able to use any MySQL client.

**Reply by billy1624** [reply]
Thanks! @lance6716 But is there any ORMs that support TiDB specific features instead of just providing MySQL compatible features?

**Reply by Icemap** [reply]
Sorry @billy1624, we don't have an ORM support specific features of TiDB in Rust now. You can use any MySQL driver or ORM instead. But we are improving the adaptability of ORM in the TiDB ecosystem. We currently support Hibernate ORM in the Java language.
Do you have any recommendations for ORM in the Rust language? We will investigate and consider adapting it in the future.

**Reply by billy1624** [reply]
Hey @Icemap, thanks for the reply! I can't find any documentation about TiDB specific features that are available in Hibernate ORM. I might overlook something though. Can you name a few features?

Btw... our team, [SeaQL](https://github.com/SeaQL), is extending [SeaORM](https://github.com/SeaQL/sea-orm) to support TiDB. And, it's actually a project under [Google Summer of Code 2022](https://summerofcode.withgoogle.com/programs/2022/projects/MebkaZS7).

**Reply by Icemap** [reply]
Hi @billy1624, I already had a conversation with @itwaix from your team.
We only have this [mysql-compatibility page](https://docs.pingcap.com/tidb/stable/mysql-compatibility) now. You can use [developer guide](https://docs.pingcap.com/zh/tidb/stable/dev-guide-overview) to see some of specific features, such as: [follower read](https://docs.pingcap.com/tidb/stable/dev-guide-use-follower-read), [stale read](https://docs.pingcap.com/tidb/stable/dev-guide-use-stale-read), [optimistic transaction](https://docs.pingcap.com/tidb/stable/dev-guide-optimistic-and-pessimistic-transaction), [sequence](https://docs.pingcap.com/tidb/stable/sql-statement-create-sequence#create-sequence).

**Reply by Oreoxmt** [reply]
Hi @billy1624, for more details about TiDB specific functions, see [TiDB Specific Functions](https://docs.pingcap.com/tidb/stable/tidb-functions). You can search for the keyword "TiDB extension to MySQL syntax" in [Docs site](https://docs.pingcap.com) to know some SQL statements which are [TiDB extension to MySQL syntax](https://docs.pingcap.com/search?type=tidb&version=stable&q=a%20TiDB%20extension%20to%20MySQL%20syntax), such as [`ADMIN`](https://docs.pingcap.com/tidb/stable/sql-statement-admin#admin), [`ALTER PLACEMENT POLICY`](https://docs.pingcap.com/tidb/stable/sql-statement-alter-placement-policy#alter-placement-policy), [`BACKUP`](https://docs.pingcap.com/tidb/stable/sql-statement-backup#backup), [`CHANGE PUMP`](https://docs.pingcap.com/tidb/stable/sql-statement-change-pump#change-pump).

**Reply by billy1624** [reply]
Thanks! @Icemap @Oreoxmt We'll keep you posted! And, perhaps invite you guys for a early beta once it's in shape :)
